src/components/ui/input.tsx

```diff
@@ -1,18 +1,31 @@
+import { VariantProps, cva } from "class-variance-authority";
 import * as React from "react";
 
 import { cn } from "~/lib/utils";
 
-export type InputProps = React.InputHTMLAttributes<HTMLInputElement>;
+export interface InputProps
+  extends React.InputHTMLAttributes<HTMLInputElement>,
+    VariantProps<typeof inputVariants> {}
+
+const inputVariants = cva(
+  "flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50",
+  {
+    variants: {
+      variant: {
+        defaut: "",
+        primary:
+          "rounded-md bg-primary-50 px-5 py-2 text-black outline outline-2 outline-primary-300 hover:bg-primary-200",
```

Comment:
```suggestion
          "rounded-md bg-primary-50 px-5 py-2 text-black outline outline-2 outline-primary-300",
```

the hover is weird LOL

Reply:
not a blocker tho